tools/test-documentation

```diff
@@ -1,92 +1,110 @@
-#!/usr/bin/env bash
-set -e
-
-color_message() {
-    local color_code="$1" message="$2"
-    printf '\e[%sm%s\e[0m\n' "$color_code" "$message" >&2
-}
-
-loglevel=()
-
-usage() {
-    cat <<EOF
-usage:
-  --help, -h                   show this help message and exit
-  --loglevel=LEVEL, -L LEVEL   log level (default: ERROR)
-  --skip-check-links           skip checking of links
-  --skip-external-links        skip checking of external links
-EOF
-}
-
-args="$(getopt -o hL: --long help,loglevel:,skip-check-links,skip-external-links -- "$@")" \
-    || {
-        usage >&2
-        exit 1
-    }
-eval "set -- $args"
-while true; do
-    case "$1" in
-        -h | --help)
-            usage
-            exit 0
-            ;;
-        -L | --loglevel)
-            loglevel=("$1" "$2")
-            shift 2
-            ;;
-        --skip-check-links)
-            skip_check_links=1
-            shift
-            ;;
-        --skip-external-links)
-            skip_external_links=1
-            shift
-            ;;
-        --)
-            shift
-            break
-            ;;
-        *) exit 1 ;;
-    esac
-done
-
-cd "$(dirname "$0")"/../docs
+#!/usr/bin/env python3
+import argparse
+import os
+import subprocess
+import sys
+from typing import Sequence
+
+ZULIP_PATH = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+os.chdir(ZULIP_PATH)
+sys.path.insert(0, ZULIP_PATH)
+
+# check for the venv
+from tools.lib import sanity_check
+
+sanity_check.check_venv(__file__)
+
+
+def color_message(color_code: str, message: str) -> None:
+    print(f"\033[0;{color_code}m{message}\033[0m", file=sys.stderr)
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "-L",
+    "--loglevel",
+    dest="loglevel",
+    nargs=1,
+    help="Log level (default: ERROR)",
+)
+parser.add_argument(
+    "--skip-check-links",
+    dest="skip_check_links",
+    action="store_true",
+    help="Skip checking of links.",
+)
+parser.add_argument(
+    "--skip-external-links",
+    dest="skip_external_link_check",
+    # default = false
+    action="store_true",
+    help="Skip checking of external links.",
+)
+options = parser.parse_args()
+# loglevel should be used as an extra command line option
+# in the form of ["-L", "LOGLEVEL"]
+# if loglevel is defined, insert "-L" ahead of loglevel
+if options.loglevel:
+    options.loglevel.insert(0, "-L")
+# if loglevel is undefined, make loglevel an empty list
+else:
+    options.loglevel = []
```

Comment:
This variable might be better defaulted as a new `loglevel_options` variable, rather than mutating `options`?

Reply:
yes, i will fix it in a later check in.